### DISCUSSION TITLE
Feature request: Implement shutter/blind

### DISCUSSION THREAD

**Discussion by stefanbode** [jziolkowski/tdm#48] [Ideas]
xrv27 adds support for shutters with many new options. Would be great if we can integrate this here. Let me know, if you need support. For commands search wikipages "blinds".

**Comment by jziolkowski** [comment]
I'll look into it.

Can you paste some MQTT responses from such commands? Because for this to work, TDM needs some feedback.

**Comment by stefanbode** [comment]
shutterrelay1 .. 4 -> Number ( Number of the first relay for the shutter 1..4, if = 0, then shutter is DISABLED)
shutteropenduration1..4 -> float Number (number of seconds eg 22.5 to open shutter)
shuttercloseduration1..4 -> float Number (number of seconds eg 22.5 to close shutter)
shuttersethalfway1..4 -> Number (value 1..100 of position where shutter is half open)

**Comment by meingraham** [comment]
Stefan,

Do you have samples of the JSON payloads for those scenarios?

**Comment by stefanbode** [comment]
Hi, there is no JSON reply. These commands reply with a pure number.

**Comment by jziolkowski** [comment]
For the response to be usable it must either reply on /RESULT topic with correct JSON payload ({"<cmd_name>":"\<value\>"}) or on its dedicated response topic.

Tasmota reply to command POWER is prime example.

Bare number returned to RESULT topic is useless.

**Comment by stefanbode** [comment]
Ok, great. Let me work on the shutter software before bringing it to tdm

**Comment by jziolkowski** [comment]
Great. If you copy the RESULT behavior of the POWER command then it will work out of the box. Also, if possible, add commands which return settings of all related functions in one json payload (for example, Var PulseTime or RuleTimer commands issued without index or and parameters return the current values of related settings 1..n)

**Comment by jziolkowski** [comment]
done some time ago, works in the latest release
